### PR TITLE
TILA-2504: Add reservation time modification modal

### PR DESCRIPTION
**Pull request by joonatank**
- Edit time form, schema and mutation.
- Edit Time Page replaced with a modal.
- DateInput changed to use strings instead of dates because keyboard / default value issues.

### CODE REVIEW

File: admin-ui/src/component/my-units/MyUnitRecurringReservation/MyUnitRecurringReservationForm.tsx [hunk 1]
```diff
@@ -210,8 +210,8 @@ const MyUnitRecurringReservationForm = ({ reservationUnits }: Props) => {
   const checkedReservations = useFilteredReservationList({
     items: newReservations.reservations,
     reservationUnitPk: reservationUnit?.pk ?? undefined,
-    begin: getValues("startingDate"),
-    end: getValues("endingDate"),
+    begin: parse(getValues("startingDate"), "dd.MM.yyyy", new Date()),
+    end: parse(getValues("endingDate"), "dd.MM.yyyy", new Date()),
```
Comment (kieferbonk) [inline admin-ui/src/component/my-units/MyUnitRecurringReservation/MyUnitRecurringReservationForm.tsx:214]: There seems to be a lot of repetition in parsing and formatting the dates. A general helper function could ease readability.

File: admin-ui/src/component/reservations/requested/hooks/index.ts [hunk 2]
```diff
@@ -22,6 +23,29 @@ import { GQL_MAX_RESULTS_PER_QUERY } from "../../../../common/const";
 
 export { default as usePermission } from "./usePermission";
 
+const getEventName = (eventType?: string, title?: string) =>
+  eventType === "blocked" ? "Suljettu" : title?.trim();
```
Comment (kieferbonk) [inline admin-ui/src/component/reservations/requested/hooks/index.ts:27]: Might want to get the string from translations.

File: admin-ui/src/component/reservations/requested/hooks/index.ts [hunk 3]
```diff
@@ -22,6 +23,29 @@ import { GQL_MAX_RESULTS_PER_QUERY } from "../../../../common/const";
...
+
+const getReservationTitle = (r: ReservationType) => r.reserveeName ?? "";
+
+const convertReservationToCalendarEvent = (r: ReservationType) => ({
+  title: getEventName(r.type ?? undefined, getReservationTitle(r)),
+  event: {
+    ...r,
+    name: r.name?.trim() !== "" ? r.name : "No name",
```
Comment (kieferbonk) [inline admin-ui/src/component/reservations/requested/hooks/index.ts:35]: Might want to get the string from translations. Also fi/en?

**Review by kieferbonk — CHANGES_REQUESTED** [review]

File: admin-ui/src/component/reservations/requested/hooks/index.ts (hunk 3 above)
Comment (joonatank) [reply admin-ui/src/component/reservations/requested/hooks/index.ts:35]: This is fixed in another PR. it should be empty

**Review by kieferbonk — APPROVED** [review]

**Pull request merged**